src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala

```diff
@@ -248,20 +248,22 @@ final case class PurposeApiServiceImpl(
 
     val result: Future[Purpose] = for {
       organizationId <- getOrganizationIdFutureUUID(contexts)
-      purposeUUID    <- purposeId.toFutureUUID
-      _              <-
+      eService       <- catalogManagementService.getEServiceById(purposeUpdateContent.eserviceId)
+      _           <- if (eService.mode == Deliver) Future.failed(EServiceNotInDeliverMode(eService.id)) else Future.unit
```

Comment:
We should inhibit the update of a purpose of an eservice in `Receive` mode, right?
```suggestion
      _           <- if (eService.mode == Receive) Future.failed(EServiceNotInDeliverMode(eService.id)) else Future.unit
```